### PR TITLE
Annotation Query Fix

### PR DESCRIPTION
**Pull request by thehabes**
See #7

### CODE REVIEW

File: js/deer-utils.js [hunk 1]
```diff
@@ -167,11 +167,31 @@ export default {
...
      * @param {String} id URI for the targeted entity
+     * @param [String] targetStyle other formats of resource targeting.  May be null
      */
-    findByTargetId: async function (id) {
+    findByTargetId: async function (id, targetStyle) {
         let everything = Object.keys(localStorage).map(k => JSON.parse(localStorage.getItem(k)))
-        let obj = {
-            target: id
+        //Backwards compatible without the new parameter
+        if(targetStyle === undefined){
```
Comment (cubap) [inline js/deer-utils.js]: Prefer to use a constructor in the parameter `function (id, targetStyle=[])`

File: js/deer-utils.js [hunk 2]
```diff
@@ -167,11 +167,31 @@ export default {
      * Execute query for any annotations in RERUM which target the
      * id passed in. Promise resolves to an array of annotations.
      * @param {String} id URI for the targeted entity
+     * @param [String] targetStyle other formats of resource targeting.  May be null
      */
-    findByTargetId: async function (id) {
+    findByTargetId: async function (id, targetStyle) {
```
Comment (cubap) [inline js/deer-utils.js]: Prefer to use a constructor in the parameter `function (id, targetStyle=[])`

File: js/deer-utils.js [hunk 3]
```diff
@@ -167,11 +167,31 @@ export default {
...
         let everything = Object.keys(localStorage).map(k => JSON.parse(localStorage.getItem(k)))
-        let obj = {
-            target: id
+        //Backwards compatible without the new parameter
+        if(targetStyle === undefined){
+            targetStyle = []
+        }
+        //Users pass the craziest things...
+        if (!Array.isArray(targetStyle)) {
+            if(typeof targetStyle === "string"){
```
Comment (cubap) [inline js/deer-utils.js]: You already discard non-strings later on, so you can just fire the next line regardless.

File: js/deer-utils.js [hunk 4]
```diff
@@ -167,11 +167,31 @@ export default {
...
+                targetStyle = [targetStyle]
+            }
+            else{
```
Comment (cubap) [inline js/deer-utils.js]: The comment is fine, but `targetStyle` here may be an object or number, so that can be dropped into an array without breaking anything. I wouldn't blank an Array without at least noting the major error in what the dev intended to send in and what is being processed.

tl;dr : remove everything but the original `if`

File: js/deer-utils.js [hunk 5]
```diff
@@ -167,11 +167,31 @@ export default {
...
+                //certainly your key isn't a number.  keys arent objects.  what are you doing?
+                targetStyle = []
+            }
+        }
+        targetStyle = targetStyle.concat(["target", "target.@id", "target.id"]) //target.source?
```
Comment (cubap) [inline js/deer-utils.js]: Does `target.@id` work? It doesn't require `[]` like in true JS?

File: js/deer-utils.js [hunk 6]
```diff
@@ -167,11 +167,31 @@ export default {
...
+        let obj = {"$or":[]}
+        for (let target of targetStyle) {
+            //If you passed an Array and it wasn't a Array of strings, I don't trust you.  Ignore.
```
Comment (cubap) [inline js/deer-utils.js]: specifically, just ignore that one entry.

File: js/deer-utils.js (hunk 5 above)
Comment (thehabes) [reply js/deer-utils.js]: Yes.  mongo querying does not support bracket notation.  They only characters that can't be used in `key` names are `$` and `.` so that this is supported across special characters.


fields

**Comment by thehabes** [comment]
Merged in deer-LR, clean merge no problems.

________________________________
